### DISCUSSION TITLE
Is it possible to force UNIQUE constraint over some property?

### DISCUSSION THREAD

**Discussion by AntiTenzor** [usinesoft/Cachalot#25] [Q&A] [answered]
For example, I have a class with integer Primary Key   **ExchangeID** 
```
    [Client.Interface.ServerSideValue(Client.Core.IndexType.Primary)]
    public int ExchangeID { get; private set; }
```

At the same time this class has string property   **Code**   which is not a primary key,
but **it must be unique** in the cache server (or even at cache cluster) as well as in the backing database.
```
    [Client.Interface.ServerSideValue(Client.Core.IndexType.Dictionary)]
    public string Code { get; private set; }
```

Is it possible to tell Cachalot server to force this constraint
over a collection of similar objects in single DataSource?

**Comment by usinesoft** [comment, marked as answer]
In the old versions there was a unique constraint.
I removed it as I realized in a distributed cluster it is close to impossible to enforce it in an efficient way.
A PUT operation is now touching one node in a cluster. And I prefer it to stay this way

**Comment by usinesoft** [comment]
The unique constraint may be enforced at a higher level. In your client application

**Comment by AntiTenzor** [comment]
In distributed system different utilities that work with DB/Cache may have it's own logic and its own way of creating similar objects.

In my practice too much DB constraints is a hell, but too few -- is a mess. =)
